### PR TITLE
Migrated `domainzonefile` datasource to Framework

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Migrated `domainzonefile` datasource to Framework.

## ✔️ How to Test

`make PKG_NAME="linode/domainzonefile" testacc`

Ticket: TPT-1975

### CODE REVIEW

**Comment by ezilber-akamai** [comment]
/acctest sha=4a1e7402f54df04db572833c55ef32d8f391341c pkg=linode/domainzonefile

File: linode/domainzonefile/framework_datasource.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{}
+}
+
+type DataSource struct {
+	meta *helper.FrameworkProviderMeta
```
Comment (yec-akamai) [inline linode/domainzonefile/framework_datasource.go:20]: why don't we use `client *linodego.Client` here as we normally do?

Comment (ezilber-akamai) [reply linode/domainzonefile/framework_datasource.go:20]: The client is already defined in the `helper.FrameworkProviderMeta` so I removed it to reduce redundancy.

**Review by yec-akamai — APPROVED** [review]
Looks good!

**Review by lgarber-akamai — APPROVED** [review]

File: linode/domainzonefile/framework_datasource_schema.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+var frameworkDatasourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"domain_id": schema.Int64Attribute{
+			Description: "The domain's ID.",
+			Required:    true,
+		},
+		"zone_file": schema.ListAttribute{
+			Description: "Lines of the zone file for the last rendered zone for this domain.",
+			Optional:    true,
```
Comment (lgarber-akamai) [inline linode/domainzonefile/framework_datasource_schema.go:16]: This field might need to be computed rather than optional 👍

Comment (ezilber-akamai) [reply linode/domainzonefile/framework_datasource_schema.go]: Fixed!

**Comment by ezilber-akamai** [comment]
/acctest sha=369975021e83fa12a76e162c2bc9d996bb08d3e7 pkg=linode/domainzonefile

**Pull request merged**